code/modules/merchant/merchant_programs.dm

```diff
@@ -81,6 +81,26 @@
 			bank -= response
 		return
 	last_comms = "PAD NOT CONNECTED"
+	
+/datum/computer_file/program/merchant/proc/bulk_offer(var/datum/trader/T, var/num)
+	var/BulkAmount = input("How many items? (1-50)") as num
```

Comment:
You could make it accept 0 as cancelling input. In that case the loop will just not execute and you just need to change line 90 if check to be `if(BulkAmount < 0 || BulkAmount > 50)`

Reply:
Implemented. Resolving